eth2near/ethashproof/dag_merkle_root.go

```diff
@@ -46,10 +46,16 @@ func processDuringRead(f *os.File, startIn128Res int, fullSizeIn128Res uint32, m
 // 3. If saveCache is true, save root merkle tree of 10 levels
 //    to disk
 // 4. Return merkle root
-func CalculateDatasetMerkleRoot(epoch uint64, saveCache bool) (mtree.Hash, error) {
+func CalculateDatasetMerkleRoot(epoch uint64, saveCache bool, cacheDir string) (mtree.Hash, error) {
 	blockno := epoch * 30000
 	fmt.Printf("Make the dag\n")
-	ethash.MakeDAG(blockno, ethash.DefaultDir)
+
+	if cacheDir == "|default|" {
```

Comment:
Please don't use string literal as an option, you can just check for empty string here.